newIDE/app/src/Course/VideoBasedCourseChapterView.js

```diff
@@ -0,0 +1,231 @@
+// @flow
+
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+
+import type {
+  VideoBasedCourseChapter,
+  CourseChapter,
+} from '../Utils/GDevelopServices/Asset';
+import Text from '../UI/Text';
+import { ColumnStackLayout } from '../UI/Layout';
+import Paper from '../UI/Paper';
+import RaisedButton from '../UI/RaisedButton';
+import { Line, Spacer } from '../UI/Grid';
+import GDevelopThemeContext from '../UI/Theme/GDevelopThemeContext';
+import Divider from '@material-ui/core/Divider';
+import FlatButton from '../UI/FlatButton';
+import ChevronArrowBottom from '../UI/CustomSvgIcons/ChevronArrowBottom';
+import ChevronArrowRight from '../UI/CustomSvgIcons/ChevronArrowRight';
+import Cloud from '../UI/CustomSvgIcons/Cloud';
+import VideoBasedCourseChapterTaskItem from './VideoBasedCourseChapterTaskItem';
+import { useResponsiveWindowSize } from '../UI/Responsive/ResponsiveWindowMeasurer';
+import PreferencesContext from '../MainFrame/Preferences/PreferencesContext';
+import { rankLabel } from '../Utils/Ordinal';
+import type { CourseChapterCompletion } from '../MainFrame/EditorContainers/HomePage/UseCourses';
+import LockedCourseChapterPreview from './LockedCourseChapterPreview';
+import CourseChapterTitle from './CourseChapterTitle';
+
+const getYoutubeVideoIdFromUrl = (youtubeUrl: ?string): ?string => {
```

Comment:
this could become go into a utils file (you're using it twice)

Reply:
Good idea, thanks!